src/proc/occlusion-filter.cpp

```diff
@@ -45,51 +81,116 @@ namespace librealsense
+                        if (points_ptr->z)
+                        {
+                            //Occlusion detection
+                            if (pixels_ptr->x < maxInLine || (pixels_ptr->x == maxInLine && (points_ptr->z - maxZ) > occZTh))
+                            {
+                                *points_ptr = { 0, 0, 0 };
+                                occDilationLeft = occDilationSz;
+                            }
+                            else
+                            {
+                                maxInLine = pixels_ptr->x;
+                                maxZ = points_ptr->z;
+                                if (occDilationLeft > 0)
+                                {
+                                    *points_ptr = { 0, 0, 0 };
+                                    occDilationLeft--;
+                                }
+                            }
+                        }
+                        ++points_ptr;
+                        ++uv_map_ptr;
+                        ++pixels_ptr;
                     }
-                    else
+                }
+            }
+        }
+        else if (_occlusion_scanning == vertical)
+        {
+            depth_planes[0] = (byte*)alloc.allocate(depth.get_bytes_per_pixel() * frame_size);
+            int bpp = depth.get_bytes_per_pixel();
+            rotate_image_counterclockwise<2>(depth_planes, (const byte*)(depth.get_data()), points_width, points_height);
+
+            // scan depth frame after clockwise rotation: check if there is a significant jump between adjacen pixels in Z-axis (depth), it means there could be occlusion.
+            // scan over a small window so that points depth is the same
+            // save suspected points and run occlusion-invalidation vertical scan only on them
+            // after rotation : height = points_width , width = points_height
+
+            auto rotated_depth_width = _depth_intrinsics->height;
+            auto rotated_depth_height = _depth_intrinsics->width;
+
+            for (int i = 0; i < rotated_depth_height; i++)
+            {
+                for (int j = 0; j < rotated_depth_width - VERTICAL_SCAN_WINDOW_SIZE; j = j + VERTICAL_SCAN_WINDOW_SIZE)
+                {
+                    // before rotation: occlusion detected in the positive direction of Y
+                    // after counterclockwise rotation : scan from right to left (positive direction of X) to detect occlusion 
+                    // compare depth each pixel only with the pixel on its right (i,j+1) 
+                    auto index = i * rotated_depth_width + j;
+                    auto uv_i = j;
+                    auto uv_j = rotated_depth_height - i;
+                    auto uv_index = uv_i * rotated_depth_height + uv_j; // 90 degrees rotation transform from rotated depth
+                    auto index_right = index + 1;
+                    float diff_right = abs((depth_planes[0])[index] - (depth_planes[0])[index_right]);
```

Comment:
instead of comparing floats, compare `uint16_t`-s. Bulk operations on ints is measurably faster

Reply:
I changed to uint16_t and didn't notice impact on performance